portal-impl/src/com/liferay/portal/servlet/filters/language/LanguageFilter.java

```diff
@@ -119,8 +153,115 @@ protected String translateResponse(
 			locale, content);
 	}
 
+	private String _getETagKey(HttpServletRequest httpServletRequest) {
+		return LanguageUtil.getLanguageId(httpServletRequest) +
+			StringPool.POUND + httpServletRequest.getRequestURI();
+	}
+
 	private static final Log _log = LogFactoryUtil.getLog(LanguageFilter.class);
 
+	private final ConcurrentHashMap<String, String> _eTags =
+		new ConcurrentHashMap<>();
 	private PortletConfig _portletConfig;
 
+	private static class NoCacheHttpServletRequestWrapper
```

Comment:
Is the purpose of this class to deactivate ETagFilter only?

Reply:
No, the purpose is to fool Equinox [ResourceServlet](https://github.com/eclipse-equinox/equinox/blob/master/bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/internal/servlet/ResourceServlet.java#L87-L99) so that it thinks it should not return `SC_NOT_MODIFIED` under any conditions.

Reply:
It's the only way to make it return the proper content so that we can recompute the ETag again.

Reply:
Ok I was suspecting a little bit about this thing, because deactivating a filter is easier than this. Good, thanks for clarifying